rest_framework/serializers.py

```diff
@@ -1002,8 +1003,18 @@ def get_fields(self):
                     field_cls = self._get_nested_class(depth, relation_info)
                     kwargs = get_nested_relation_kwargs(relation_info)
                 else:
-                    field_cls = self._related_class
                     kwargs = get_relation_kwargs(field_name, relation_info)
+                    to_field = kwargs.get('to_field', False)
```

Comment:
Can you just do `kwargs.pop('to_field', False` here instead of getting it, and then popping it on the next line?